packages/p2p/src/stores/order-store.js

```diff
@@ -171,6 +195,76 @@ export default class OrderStore {
         this.hideDetails(false);
     }
 
+    @action.bound
+    setOrderRating(id) {
+        const rating = this.rating_value / 20;
+
+        requestWS({
+            p2p_order_review: 1,
+            order_id: id,
+            rating,
+            ...(this.is_recommended === undefined ? {} : { recommended: this.is_recommended }),
+        }).then(response => {
+            if (response.error) {
+                this.setErrorMessage(response.error.message);
+            }
+            this.setIsRatingModalOpen(false);
+        });
+    }
+
+    @action.bound
+    subscribeToCurrentOrder() {
+        this.order_info_subscription = subscribeWS(
+            {
+                p2p_order_info: 1,
+                id: this.order_id,
+                subscribe: 1,
+            },
+            [this.setOrderDetails]
+        );
+    }
+
+    @action.bound
+    syncOrder(p2p_order_info) {
+        const { general_store } = this.root_store;
+
+        const get_order_status = createExtendedOrderDetails(
+            p2p_order_info,
+            general_store.client.loginid,
+            general_store.props.server_time
+        );
+
+        const order_idx = this.orders.findIndex(order => order.id === p2p_order_info.id);
+
+        if (this.order_id === null) {
```

Comment:
do we need to check for null? couldn't we do 

```
!this.order_id
```

Reply:
We're only checking for null status i.e., initial status.